### PR TITLE
util/memory: migrate test-infra to testify

### LINKED ISSUE #26979: migrate test-infra to testify for `util/memory` pkg

### PR DESCRIPTION
**Pull request by unconsolable**
### What problem does this PR solve?

Issue Number: close #26979 

### Release note 

```release-note
None
```

### CODE REVIEW

File: util/memory/tracker_test.go [hunk 1]
```diff
@@ -15,61 +15,40 @@ package memory
...
-type testSuite struct{}
-
-func (s *testSuite) SetUpSuite(c *C)    {}
-func (s *testSuite) TearDownSuite(c *C) {}
-func (s *testSuite) SetUpTest(c *C)     { testleak.BeforeTest() }
-func (s *testSuite) TearDownTest(c *C)  { testleak.AfterTest(c)() }
-
-func (s *testSuite) TestSetLabel(c *C) {
+func TestSetLabel(t *testing.T) {
 	tracker := NewTracker(1, -1)
```
Comment (feitian124) [inline util/memory/tracker_test.go:32]: i thought you should add `t.Parallel()` here to make tests as parallel as possible.
the same to other methods.

Comment (feitian124) [reply util/memory/tracker_test.go:34]: IMO no need blank line here

Comment (unconsolable) [reply util/memory/tracker_test.go:34]: > IMO no need blank line here

There are a few tests with blank line after `t.Parallel()`, maybe it is not necessary.

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your contribution!

**Comment by tisonkun** [comment]
@unconsolable release note is for user facing changes. Code refactor is often unrelated. Changes to none release note.

**Review by mmyj — APPROVED** [review]
lgtm

File: util/memory/tracker_test.go [hunk 2]
```diff
@@ -430,29 +429,23 @@ func (s *testSuite) TestFormatBytesWithPrune(c *C) {
 		{"9.15999984741211 MB", "9.16 MB"},
 	}
 	for _, ca := range cases {
-		b, err := s.parseByte(ca.b)
-		c.Assert(err, IsNil)
+		b, err := parseByte(ca.b)
+		require.NoError(t, err)
 		result := FormatBytes(b)
-		c.Assert(result, Equals, ca.s, Commentf("input: %v", ca.b))
+		require.Equal(t, ca.s, result)
```
Comment (xhebox) [inline util/memory/tracker_test.go:435]: Use `Equalf` to contain `fmt.Printf("input: %v\n", ca.b)` maybe.

Comment (xhebox) [reply util/memory/tracker_test.go]: Then you can remove `fmt.Print`, which will output something whatever the result is. It is annoying.

Comment (unconsolable) [reply util/memory/tracker_test.go]: I regarded `Commentf` as `fmt.Printf` previously, I have just check the document of `go-check` and found that it won't print until the assert is false. Sorry for my misunderstanding.

**Review by xhebox — APPROVED** [review]

File: util/memory/tracker_test.go (hunk 2 above)
Comment (unconsolable) [reply util/memory/tracker_test.go]: Test fails in `tidbTestSerialSuite.TestTLS` twice, I try to remove `t.Parallel()` in concurrent test `TestConsume` to eliminate it.

**Pull request merged**